### PR TITLE
[CI] Create doc issue workflow

### PR DESCRIPTION
**Pull request by kavilla**
### Description
Add workflow to create documentation issue in
https://github.com/opensearch-project/documentation-website

Update CONTRIBUTING doc on how to use.

Signed-off-by: Kawika Avilla <kavilla414@gmail.com>
 
### Issues Resolved
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1493
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
    - [ ] `yarn test:jest`
    - [ ] `yarn test:jest_integration`
    - [ ] `yarn test:ftr`
- [ ] New functionality has been documented.
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by kavilla** [comment]
@VachaShah, does https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/create-documentation-issue.yml#L28 work? I pulled the `checkout` step and `echo to markdown step` and without providing the `.` before `github` it failed so I'd imagine it'd be the same for `.ci`

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -92,7 +92,7 @@ We deeply appreciate everyone who takes the time to make a contribution. We will
...
+If we accept the PR, a [maintainer](MAINTAINERS.md) will merge your change and take care of backporting it to the appropriate branches ourselves. For the backporting process, once the PR is merged a [maintainer](MAINTAINERS.md) will label it with the appropriate target branch then the backport workflow will do the rest. For example, the `backport 1.x` label will be added to the PR that is to be backported to `1.x`. The backport branches are named in the form `backport/backport-<original PR number>-to-<base>`. These branches will be cleaned up by an auto delete workflow once the backport PR is merged. If documentation is required, a [maintainer](MAINTAINERS.md) will label the PR with the `needs-documentation` label which will then create an issue in the [Documentation Website repo](https://github.com/opensearch-project/documentation-website).
```
Comment (ashwin-pc) [inline CONTRIBUTING.md:95]: We should also call it out that it is their responsibility to document and close that issue on `documentation-website`

**Review by ashwin-pc — APPROVED** [review]

**Comment by ashwin-pc** [comment]
Nice!

**Review by ananzh — APPROVED** [review]
❤️

**Comment by joshuarrrr** [comment]
How does this work in the case where a contributor creates a parallel documentation PR at the same time? Maintainers just skip adding the label, because the issue isn't needed?

**Pull request merged**